modules/http/http_filters.c

```diff
@@ -420,224 +94,92 @@ apr_status_t ap_http_filter(ap_filter_t *f, apr_bucket_brigade *b,
-            f->r->status = HTTP_CONTINUE;
-            f->r->status_line = NULL;
-            ap_send_interim_response(f->r, 0);
-            AP_DEBUG_ASSERT(!f->r->expecting_100);
-            f->r->status_line = saved_status_line;
-            f->r->status = saved_status;
+            int saved_status = r->status;
+            const char *saved_status_line = r->status_line;
+            ap_log_rerror(APLOG_MARK, APLOG_TRACE3, 0, r,
+                          "ap_http_in_filter: sending 100-continue");
+            r->status = HTTP_CONTINUE;
+            r->status_line = NULL;
+            ap_send_interim_response(r, 0);
+            AP_DEBUG_ASSERT(!r->expecting_100);
+            r->status_line = saved_status_line;
+            r->status = saved_status;
         }
         else {
             /* https://tools.ietf.org/html/rfc7231#section-5.1.1
              *   A server MAY omit sending a 100 (Continue) response if it
              *   has already received some or all of the message body for
              *   the corresponding request [...]
              */
-            ap_log_rerror(APLOG_MARK, APLOG_DEBUG, 0, f->r, APLOGNO(10260)
+            ap_log_rerror(APLOG_MARK, APLOG_DEBUG, 0, r, APLOGNO(10260)
                           "request body already/partly received while "
                           "100-continue is expected, omit sending interim "
                           "response");
-            f->r->expecting_100 = 0;
+            r->expecting_100 = 0;
         }
     }
 
     /* sanity check in case we're read twice */
     if (ctx->at_eos) {
         e = apr_bucket_eos_create(f->c->bucket_alloc);
         APR_BRIGADE_INSERT_TAIL(b, e);
-        return APR_SUCCESS;
-    }
-
-    do {
-        apr_brigade_cleanup(b);
-        again = 0; /* until further notice */
-
-        /* read and handle the brigade */
-        switch (ctx->state) {
-        case BODY_CHUNK:
-        case BODY_CHUNK_PART:
-        case BODY_CHUNK_EXT:
-        case BODY_CHUNK_CR:
-        case BODY_CHUNK_LF:
-        case BODY_CHUNK_END:
-        case BODY_CHUNK_END_LF: {
-
-            rv = ap_get_brigade(f->next, b, AP_MODE_GETLINE, block, 0);
-
-            /* for timeout */
-            if (block == APR_NONBLOCK_READ
-                    && ((rv == APR_SUCCESS && APR_BRIGADE_EMPTY(b))
-                            || (APR_STATUS_IS_EAGAIN(rv)))) {
-                return APR_EAGAIN;
-            }
-
-            if (rv == APR_EOF) {
-                return APR_INCOMPLETE;
-            }
-
-            if (rv != APR_SUCCESS) {
-                return rv;
+        rv = APR_SUCCESS;
+        goto cleanup;
+    }
+
+    rv = ap_get_brigade(f->next, b, mode, block, readbytes);
+    if (APR_SUCCESS == rv) {
+        for (e = APR_BRIGADE_FIRST(b);
+             e != APR_BRIGADE_SENTINEL(b);
+             e = next)
+        {
+            next = APR_BUCKET_NEXT(e);
+            if (!APR_BUCKET_IS_METADATA(e)) {
+                ctx->seen_data = (e->length != 0);
```

Comment:
This may overwrite seen_data, maybe:
```
if (e->length != 0) {
    ctx->seen_data = 1;
}
```
?

Reply:
or `|=`..

Reply:
Updated PR with change.